test_communication/test/subscriber_py.py

```diff
@@ -62,7 +62,7 @@ def listener(message_name, namespace):
     spin_count = 1
     print('subscriber: beginning loop')
     while (rclpy.ok() and len(received_messages) != len(expected_msgs)):
```

Comment:
This condition will still not be met, regardless of `spin_once` timeout, and this will still be waiting forever (or to the test to timeout). So, changing `spin_once` behavior from blocking to anything else just makes the loop more busy.

I think the original test case was ok, we can change the timeout of the test to a smaller value to avoid CI taking much longer when some of these tests fail.